managed/src/main/java/com/yugabyte/yw/scheduler/Scheduler.java

```diff
@@ -249,28 +249,35 @@ private void runMultiTableBackupsTask(Schedule schedule) {
       taskParams.universeUUID, universe.name);
   }
 
-  private void runDeleteBackupTask(Backup backup, Schedule schedule) {
+  public void runDeleteBackupTask(Backup backup) {
     if (backup.state != Backup.BackupState.Completed) {
       LOG.warn("Cannot delete backup {} since it is not in completed state.",
         backup.backupUUID);
       return;
     }
     BackupTableParams backupParams = Json.fromJson(backup.backupInfo, BackupTableParams.class);
     Universe universe = null;
+    boolean isScheduled = (backup.getScheduleUUID() != null) ? true : false;
     try {
       universe = Universe.get(backupParams.universeUUID);
     } catch (Exception e) {
-      schedule.stopSchedule();
+      if (isScheduled){
+        Schedule schedule = Schedule.get(backup.getScheduleUUID());
```

Comment:
guess ` boolean isScheduled = (backup.getScheduleUUID() != null) ? true : false;` it will raiseNPE because of this condition.
`isScheduled` will be true only if `backup.getScheduleUUID()` is not null.